### PR TITLE
Add video syntax

### PR DESCRIPTION
**Pull request by fenxu**
docfx video syntax. It's first version of video syntax. Only contains link, the other properties such as width and height may should be configurable and we can modify the syntax later.

Also, we may need to extend the first version of video syntax based on users requirement.

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/DfmBlockquoteHelper.cs [hunk 1]
```diff
@@ -21,6 +21,11 @@ public static List<SplitToken> SplitBlockquoteTokens(ImmutableArray<IMarkdownTok
                     splitToken = new SplitToken(token);
                     splitTokens.Add(splitToken);
                 }
+                else if (token is DfmVideoBlockToken)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/DfmBlockquoteHelper.cs]: Can merge with the above if by adding a `|| token is ...`?

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Microsoft.DocAsCode.Dfm
+{
+    using System.Text.RegularExpressions;
+
+    using Microsoft.DocAsCode.MarkdownLite;
+
+    public class DfmVideoBlockRule : IMarkdownRule
+    {
+        private static readonly Regex _videoRegex = new Regex(@"^ *\[\!Video +(?<link>https?\:\/\/.+?) *\] *(\n|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs]: >^ *\[\!Video +(?<link>https?\:\/\/.+?) *\] *(\n|$) [](start = 64, length = 50)

Question: is it possible use syntax like GFM? [https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#youtube-videos](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#youtube-videos)

Comment (fenxu) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmVideoBlockRule.cs]: The syntax you mentioned has already been supported in our syntax.
However, That's a picture link of video syntax but not embeded video in the page. They are different.

---
In reply to: [72000890](https://github.com/dotnet/docfx/pull/526#discussion_r72000890) [](ancestors = 72000890)

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/DfmBlockquoteHelper.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ public static List<SplitToken> SplitBlockquoteTokens(ImmutableArray<IMarkdownTok
             SplitToken splitToken = null;
             foreach (var token in tokens)
             {
-                if (token is DfmSectionBlockToken || token is DfmNoteBlockToken)
+                if (token is DfmSectionBlockToken || token is DfmNoteBlockToken || token is DfmVideoBlockToken)
                 {
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/DfmBlockquoteHelper.cs]: what about as an interface?

File: src/Microsoft.DocAsCode.Dfm/DfmEngineBuilder.cs [hunk 4]
```diff
@@ -57,10 +57,11 @@ public DfmEngineBuilder(Options options, string baseDir = null) : base(options)
                 throw new ArgumentException("MarkdownNewLineBlockRule should exist!");
             }
             blockRules.Insert(index + 1, new DfmIncludeBlockRule());
-            blockRules.Insert(index + 2, new DfmYamlHeaderBlockRule());
-            blockRules.Insert(index + 3, new DfmSectionBlockRule());
-            blockRules.Insert(index + 4, new DfmFencesBlockRule());
-            blockRules.Insert(index + 5, new DfmNoteBlockRule());
+            blockRules.Insert(index + 2, new DfmVideoBlockRule());
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/DfmEngineBuilder.cs]: >Insert [](start = 23, length = 6)

InsertRange instead of Insert +1, +2, ...

**Comment by vwxyzh** [comment]
:shipit:

**Pull request merged**